unity/EditorPlugin/AssetPostprocessors/CsprojAssetPostprocessor.cs

```diff
@@ -120,13 +120,27 @@ private static bool UpgradeProjectFile(string projectFile, XDocument doc)
       changed |= SetLangVersion(projectContentElement, xmlns);
       changed |= SetProjectFlavour(projectContentElement, xmlns);
       changed |= SetManuallyDefinedCompilerSettings(projectFile, projectContentElement, xmlns);
+      changed |= SetAdditionalReference("Microsoft.CSharp.dll", projectContentElement, xmlns);
       changed |= SetXCodeDllReference("UnityEditor.iOS.Extensions.Xcode.dll", projectContentElement, xmlns);
       changed |= SetXCodeDllReference("UnityEditor.iOS.Extensions.Common.dll", projectContentElement, xmlns);
       changed |= SetDisableHandlePackageFileConflicts(projectContentElement, xmlns);
 
       return changed;
     }
 
+    private static bool SetAdditionalReference(string referenceName, XElement projectContentElement, XNamespace xmlns)
```

Comment:
It would be better to call this something like `AddMicrosoftCSharpReference` and don't bother passing the reference name, because the method already makes assumptions that it's .net 4.x only, so it's not a general purpose method.